src/components/AdvertisingSlot.js

```diff
@@ -1,39 +1,69 @@
-import React, { Component } from 'react';
-import connectToAdServer from './utils/connectToAdServer';
 import PropTypes from 'prop-types';
+import React, { useEffect, useRef, useContext } from 'react';
+import AdvertisingContext from '../AdvertisingContext';
+import calculateRootMargin from './utils/calculateRootMargin';
+import isLazyLoading from './utils/isLazyLoading';
+import getLazyLoadConfig from './utils/getLazyLoadConfig';
 
-class AdvertisingSlot extends Component {
-  componentDidMount() {
-    const { activate, id, customEventHandlers } = this.props;
-    activate(id, customEventHandlers);
-  }
-
-  componentDidUpdate(prevProps) {
-    const { activate, id, customEventHandlers } = this.props;
-    if (prevProps.activate !== activate) {
-      activate(id, customEventHandlers);
+function AdvertisingSlot({
+  id,
+  style,
+  className,
+  children,
+  customEventHandlers,
+}) {
+  const observerRef = useRef(null);
+  const containerDivRef = useRef();
+  const { activate, config } = useContext(AdvertisingContext);
+  const lazyLoadConfig = getLazyLoadConfig(config, id);
+  const isLazyLoadEnabled = isLazyLoading(lazyLoadConfig);
+  useEffect(() => {
+    if (!config || !isLazyLoadEnabled) {
+      return () => {};
```

Comment:
Why returning an empty function here?

Reply:
No wonder the code coverage decreased by 7% 😂

Reply:
ESLint rule `consistent-return`. Do you think it would be preferable to return null?

Reply:
I think it should be fine the empty function or null.
No problem with it.